### PR TITLE
Flyer Submission & Login for Organizations

### PR DESCRIPTION
**Pull request by zachseidner1**
## Overview

Adds the login flow and flyer creation for organizations.

## Changes Made

- Added the following general components:
  - VolumeTextField
  - VolumeButton
- Completed organizations login screen
- Added data store for organization slug and access code for remembering login details
- Added save login information checkbox to org login screen
- Updated GraphQL files to include mutations for flyer creation
- Created view models for organizations login screen and flyer upload screen
- Created new generalized response state class to be utilized from now on instead of individual response states for each model.
- Added file for general utility functions
- Added `letIfAllNotNull` function, which acts as a `.let` that works for multiple variables

## Test Coverage

Tested by logging into an organization and remembering login information, then uploading multiple flyers and ensuring everything was parsed correctly. Also tried not checking the box to remember login info to ensure that it would be properly forgotten.

## Screenshots & Videos
<details>
Entry point
    
Organization login screen

Flyer upload screen

Date picker

Time picker

Photo picker

Filled in flyer upload form

Published flyer success state

Newly uploaded flyer displaying on Flyers page

</details>

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/volume/ui/screens/FlyerUploadScreen.kt [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+                        text = flyerCategory,
+                        fontSize = 14.sp,
+                        color = GrayFive,
+                        fontFamily = lato
+                    )
+                }
+                DropdownMenu(
+                    expanded = categoryDropdownShowing,
+                    onDismissRequest = {
+                        Log.d(
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/android/volume/ui/screens/FlyerUploadScreen.kt]: delete this log statement

File: app/src/main/java/com/cornellappdev/android/volume/ui/screens/FlyerUploadScreen.kt [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+                            "TAG",
+                            "FlyerUploadScreen: Dismiss request"
+                        ); categoryDropdownShowing = false
+                    }) {
+                    categories.forEach { category ->
+                        DropdownMenuItem(
+                            text = { Text(text = category, fontFamily = lato, color = GrayFive) },
+                            onClick = {
+                                Log.d("x`", "FlyerUploadScreen: dropdown item clicked")
+                                categoryDropdownShowing = false
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/android/volume/ui/screens/FlyerUploadScreen.kt:453]: also delete this one

File: app/src/main/java/com/cornellappdev/android/volume/navigation/MainTabbedNavigation.kt [hunk 3]
```diff
@@ -386,5 +392,40 @@ private fun MainScreenNavigationConfigurations(
...
+        composable(route = "${Routes.UPLOAD_FLYER.route}/{organizationId}") { entry ->
+            val orgId = entry.arguments?.getString("organizationId")
+            FlyerUploadScreen(
+                organizationId = orgId ?: "",
+                onFlyerUploadSuccess = { navController.navigate(Routes.FLYER_SUCCESS.route) })
+        }
+        composable(route = "${Routes.FLYER_SUCCESS.route}", enterTransition = {
+            fadeIn(
+                initialAlpha = 0f,
+                animationSpec = tween(durationMillis = 2500)
```
Comment (emilyhu8) [inline app/src/main/java/com/cornellappdev/android/volume/navigation/MainTabbedNavigation.kt]: transition time of 2500 may be a little too long? up to you though

**Review by emilyhu8 — APPROVED** [review]
Great work! This is a really cool feature that will definitely improve the app greatly! All of the code looks very solid. I just added 2 comments about log statements.

**Comment by emilyhu8** [comment]
Just out of curiosity, how will organizations receive their access code?

**Comment by zachseidner1** [comment]
> Just out of curiosity, how will organizations receive their access code?

We are going to email organizations their access code manually.

**Pull request merged**